ext/phar/util.c

```diff
@@ -1427,14 +1427,23 @@ static int phar_hex_str(const char *digest, size_t digest_len, char **signature)
 /* }}} */
 
 #ifndef PHAR_HAVE_OPENSSL
-static int phar_call_openssl_signverify(int is_sign, php_stream *fp, zend_off_t end, char *key, size_t key_len, char **signature, size_t *signature_len, uint32_t sig_type) /* {{{ */
+static zend_result phar_call_openssl_signverify(bool is_sign, php_stream *fp, zend_off_t end, char *key, size_t key_len, char **signature, size_t *signature_len, uint32_t sig_type) /* {{{ */
 {
-	zend_fcall_info fci;
-	zend_fcall_info_cache fcc;
-	zval retval, zp[4], openssl;
+	zval retval, zp[4];
 	zend_string *str;
 
-	ZVAL_STRINGL(&openssl, is_sign ? "openssl_sign" : "openssl_verify", is_sign ? sizeof("openssl_sign")-1 : sizeof("openssl_verify")-1);
+	zend_function *fn = NULL;
+	if (is_sign) {
+		fn = zend_hash_str_find_ptr_lc(EG(function_table), ZEND_STRL("openssl_sign"));
```

Comment:
You don't need the _lc variants in this if/else, because the input string is already lowercased.

Reply:
Also, by looking into EG(function_table) instead of CG(function_table), you're allowing global user functions to be called. The previous code already allowed that too though...

Reply:
I'm happy to change it to `CG` as that's probably "better".